python-sdk/pachyderm_sdk/interceptor.py

```diff
@@ -43,4 +60,14 @@ def _check_connection_error(grpc_future: grpc.Future):
                     " python_pachyderm within the pipeline. "
                 )
             raise ConnectionError(error_message) from error
+        unable_to_serialize = "Exception serializing request" in error.details()
+        if error.code() == grpc.StatusCode.INTERNAL and unable_to_serialize:
+            error_message = (
+                "An error occurred while trying to serialize the following"
+                f" {request.__class__.__qualname__} message.\n "
+                " This is most likely due to one of the fields of"
+                " this message having a value with an incorrect type.\n"
+                f"\tMessage: {request}"
+            )
+            raise TypeError(error_message) from error
         raise AuthServiceNotActivated.try_from(error)
```

Comment:
is the error guaranteed to be this type of error at this point? might be good to validate this. if i'm understanding correctly, before this change serialization errors might have been reported as auth service not activated errors.

Reply:
Yeah I think this `try_from` error was a mistake. What it does is either returns an instantiated `AuthServiceNotActivated` error if that is what the grpc error represents or it returns the error with no change. I think since we're updating this method it makes sense to make this work like the other errors